### PR TITLE
enhance top description

### PR DESCRIPTION
**Pull request by h4ppysmile**
add a little more description and references

### CODE REVIEW

File: patterns/structural/proxy.py [hunk 1]
```diff
@@ -1,4 +1,11 @@
 """
+*What is this pattern about?
+Proxy is used in places where you want to add functionality to a class without changing its interface. The main class is called `Real Subject`. A client should use the proxy or the real subject without any code change, so both must have the same interface. Logging and controlling access to the real subject are some of the proxy pattern usages.
```
Comment (faif) [inline patterns/structural/proxy.py]: Please split this long line. You don't have to make it strictly 80 characters long, but use an acceptable length for modern monitors (maybe 120 characters).

Comment (h4ppysmile) [reply patterns/structural/proxy.py]: Thanks a lot, @faif for your feedback!
This is my first open-source commitment.

I changed lengths of line to 80 char and changed the example to clarify more.

File: patterns/structural/proxy.py [hunk 2]
```diff
@@ -1,55 +1,67 @@
...
+same interface. Logging and controlling access to the real subject are some of
+the proxy pattern usages.
+
+*References:
+https://refactoring.guru/design-patterns/proxy/python/example
+https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Fronting.html
+
 *TL;DR
-Provides an interface to resource that is expensive to duplicate.
+Add functionality to a resource without changing its interface.
```
Comment (gyermolenko) [inline patterns/structural/proxy.py]: I don't like how this phrase emphasises on adding functionality.
Although proxy can provide additional logic (caching, or checking permissions or other preconditions), but it is not important, and it is not visible for its client.

File: patterns/structural/proxy.py [hunk 3]
```diff
@@ -1,55 +1,67 @@
...
-# Sales Manager ready to talk
-# Proxy checking for Sales Manager availability
-# Sales Manager is busy
-# Proxy checking for Sales Manager availability
-# This Sales Manager will not talk to you whether he/she is busy or not
-# Proxy checking for Sales Manager availability
-# This Sales Manager will not talk to you whether he/she is busy or not
+    proxy = Proxy()
+    proxy.do_the_job('admin')
+    proxy.do_the_job('anonymous')
```
Comment (gyermolenko) [inline patterns/structural/proxy.py]: Could you please also show here that client can call RealSubject the same way as the Proxy? (the essential "interface was not changed" part of this pattern). Results would be different ofcourse.

File: patterns/structural/proxy.py [hunk 4]
```diff
@@ -1,55 +1,67 @@
...
+https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Fronting.html
+
 *TL;DR
-Provides an interface to resource that is expensive to duplicate.
+Add functionality to a resource without changing its interface.
 """
 
-import time
 
+class Subject:
```
Comment (gyermolenko) [inline patterns/structural/proxy.py]: I don't think it is necessary to have this Abstract class.

**Comment by h4ppysmile** [comment]
@gyermolenko 

all done, is it ok?

File: patterns/structural/proxy.py [hunk 5]
```diff
@@ -1,55 +1,88 @@
...
+    client(proxy, 'anonymous')
+
+    print()
+
+    print('# doing the job with real-subject:')
+    client(real_subject, 'admin')
+    client(real_subject, 'anonymous')
+
+
+OUTPUT = """
```
Comment (gyermolenko) [inline patterns/structural/proxy.py]: I would like you to try this kind of outputs testing
https://github.com/faif/python-patterns/blob/8dbfbd67b8c178b23b84e74302563a0f6ec95cfa/patterns/behavioral/template.py#L49-L72

It is an ongoing task to move all OUTPUTs to be like doctests.

**Comment by h4ppysmile** [comment]
@gyermolenko 
just did it!

**Comment by faif** [comment]
@gyermolenko Do you think that it's ready to be merged now? I'm fine with it

**Comment by gyermolenko** [comment]
LGTM :)
@h4ppysmile thanks for making the changes!
